enterprise/coderd/roles.go

```diff
@@ -27,6 +27,14 @@ func (api *API) patchRole(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if err := httpapi.NameValid(req.Name); err != nil {
```

Comment:
nit: do we need an integration test for this case?

Reply:
Added